toolchain/driver/compile_subcommand.cpp

```diff
@@ -660,6 +665,18 @@ auto CompileSubcommand::Run(DriverEnv& driver_env) -> DriverResult {
     stream_consumer.Flush();
   });
 
+  PrettyStackTraceFunction flush_on_crash([&](llvm::raw_ostream& out) {
+    // When crashing, print pending diagnostics to the crash stream.
+    // TODO: Eventually we'll want to limit the count.
+    out << "Pending diagnostics\n";
```

Comment:
Should we skip printing this if diagnostic streaming is enabled? (And maybe if there are no diagnostics?)

Reply:
PrettyStackTrace requires a [stack allocation](https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Support/PrettyStackTrace.h#L47-L51). A unique_ptr sort of works, but only if you immediately allocate it (or don't allocate it) so that destruction order is still stack-like. I believe it's infeasible to skip printing entirely: you end up with a number and malformatted stack trace output. We can print something different, but we cannot choose to skip printing based on later state.

I've added a slightly different label if streaming is enabled. We could skip both printing and flushing for streamed errors, but my leaning would be that it's potentially helpful and more consistent to flush in both modes.